### ISSUE TITLE
Nexus API returns 403 (Forbidden) when checking for updates

### ISSUE THREAD

**Issue by squid-box** [Nexus-Mods/Nexus-Mod-Manager#346]
**Describe the bug** 
When NMM tries to get information from the legacy API about what the latest NMM version is available, it only gets a 403 response.

**Environment**
- NMM version: 0.65.5
- Mod(s) used: N/A

[TraceLog20180526001952.txt](https://github.com/Nexus-Mods/Nexus-Mod-Manager/files/2040987/TraceLog20180526001952.txt)

**To Reproduce**
1. Start NMM.
2. Click _Update Nexus Mod Manager_.

**Additional information** 
The adress that is checked is https://legacy-api.nexusmods.com/NMM?GetLatestVersion - Is this a correct call, or did something change server-side?

**Comment by squid-box** [comment]
@DuskDweller: Who would we contact about this? The only thing we could do from the NMM side is pretty much to disable the update functionality (maybe that makes sense, will Nexus upload/serve new versions anytime soon?).

**Comment by neoh4x0r** [comment]
@DuskDweller 
@squid-box 

A 403 (Forbidden) is received for url: https://legacy-api.nexusmods.com/NMM?GetLatestVersion

Logins/authentication seem to be disabled (they don't work anymore?)....
~~I keep getting a **null** authentication token~~, yet can still update mods, etc
(the null auth token was due to my own custom service client.....but by using the built-in client I get a non-null token.)

~~The reason is because you must be authenticated before you can access the url.~~
~~(in fact this must be done to access any of the urls under legacy-api.nexusmods.com)~~
~~A webrequest must be created with the user's login sid sent along with the request.~~
~~It must be tied to the GatekeeperNexusREST in app.config.)~~

Currently this is what it returns ~~(once authenticated)~~:
`0.63.14|https:\/\/legacy-api.nexusmods.com\/Releases\/4.5\/Nexus Mod Manager-0.63.14.exe`

This commit is responsible for this (HTTPS Support) : 
ce7164d8df175834f3668914cc8fb10a024fdce3

NexusClient/ModRepositories/Nexus/NexusLinks.cs

```
public static string LatestVersion
{
  get
  {
    return @"https://legacy-api.nexusmods.com/NMM?GetLatestVersion";
  }
}
```

Here is some code that could fix this:

**NexusClient/ModRepositories/Nexus/INexusModRepositoryApi.cs**
```
/// <summary>
/// Gets the latest NMM version.
/// </summary>
/// <returns>The latest nmm version.</returns>
[OperationContract]
[WebInvoke(
	BodyStyle = WebMessageBodyStyle.Bare,
	UriTemplate = "NMM?GetLatestVersion",
	ResponseFormat = WebMessageFormat.Json)]
string GetLatestVersion();
```
**NexusClient/ModRepositories/IModRepository.cs**
`String GetLatestVersion();`

**NexusClient/ModRepositories/Nexus/NexusModRepository.cs**
```
#region GetLatestVersion
	/// <summary>
	/// Gets the latest NMM version
	/// </summary>
	/// <returns>The latest version.</returns>
	public string GetLatestVersion()
	{
		string version = "";
		if (IsOffline)
			return null;
		try {
			using (IDisposable dspProxy = (IDisposable)GetProxyFactory().CreateChannel())
			{
				INexusModRepositoryApi nmrApi = (INexusModRepositoryApi)dspProxy;
				version = nmrApi.GetLatestVersion();
			}
		} catch (TimeoutException e) {
			throw new RepositoryUnavailableException (String.Format ("00001 Cannot reach the {0} metadata server.", Name), e);
		} catch (CommunicationException e) {
			throw new RepositoryUnavailableException (String.Format ("00002 Cannot reach the {0} metadata server.", Name), e);
		} catch (SerializationException e) {
			throw new RepositoryUnavailableException (String.Format ("00003 Cannot reach the {0} metadata server.", Name), e);
		}
		return version;
	}
#endregion
```

**Comment by squid-box** [comment]
Playing around a bit with this I'm still only getting empty replies from the server, logged in or not.

If you're able to get it working I think everyone would welcome a Pull Request :)

**Comment by ousnius** [comment]
The legacy-api site does not exist any longer, afaik.

**Comment by squid-box** [comment]
Regardless of if it works or not, I guess there won't be any more releases through Nexus again (barring a GDPR2).

In my mind there's two options:
1. Change the updater to check for Github releases instead (eg #350).
2. Disable/Remove the updating feature all together (for NMM, not mods).

**Comment by neoh4x0r** [comment]
@ousnius 
I am using legacy-api site right now....and I can confirm that it is working.

I have been maintaining my own fork of 0.52.3 for use with wine & mono under linux.

After updating app.config and switching from nmm.nexusmods.com to legacy-api.nexusmods.com the program was able to correctly access the api.

I will have to checkout the latest version of NMM and see if I can forward-port my changes (or at least provide patches).

**Comment by neoh4x0r** [comment]
I just looked at the code and I see this file is where the latest version is queried
(this ExtendedWebClient is not making use of the service API and this is why it fails....you must be logged in)

**The code that I previously posted would fix this....**

However, the following class would need to be updated to make use of it.

PS:  with the latest version (0.65.5): 94a9bf8931f5e8630284680ec6257cfca283f696
I was able to login.

**NexusClient/Updating/ProgrammeUpdater.cs**
```
private class ExtendedWebClient : WebClient
{
	private int m_intDefaultTimeout = 30000;

	public Int32 CustomTimeout
	{
		get
		{
			return m_intDefaultTimeout;
		}
		set
		{
			m_intDefaultTimeout = value;
		}
	}

	public ExtendedWebClient() : this(30000) { }

	public ExtendedWebClient(int p_intTimeout)
	{
		this.m_intDefaultTimeout = p_intTimeout;
	}

	protected override WebRequest GetWebRequest(Uri uri)
	{
		WebRequest w = base.GetWebRequest(uri);
		w.Timeout = m_intDefaultTimeout;
		return w;
	}
}

/// <summary>
/// Gets the newest available programme version.
/// </summary>
/// <returns>The newest available programme version,
/// or 69.69.69.69 if now information could be retrieved.</returns>
private Version GetNewProgrammeVersion(out string p_strDownloadUri)
{
	ExtendedWebClient wclNewVersion = new ExtendedWebClient(15000);
	Version verNew = new Version("69.69.69.69");
	p_strDownloadUri = String.Empty;

	try
	{
		string strNewVersion = wclNewVersion.DownloadString(NexusLinks.LatestVersion);
		if (!String.IsNullOrEmpty(strNewVersion))
		{
			verNew = new Version(strNewVersion.Split('|')[0]);
			p_strDownloadUri = strNewVersion.Split('|')[1];
		}
	}
	catch (WebException)
	{
		try
		{
			string strNewVersion = wclNewVersion.DownloadString(NexusLinks.LatestVersion4dot5);
			if (!String.IsNullOrEmpty(strNewVersion))
			{
				verNew = new Version(strNewVersion.Split('|')[0]);
				p_strDownloadUri = strNewVersion.Split('|')[1];
			}
		}
		catch (WebException e)
		{
			Trace.TraceError(String.Format("Could not connect to update server: {0}", e.Message));
		}
		catch (ArgumentException e)
		{
			Trace.TraceError(String.Format("Unexpected response from the server: {0}", e.Message));
		}
	}
	catch (ArgumentException e)
	{
		Trace.TraceError(String.Format("Unexpected response from the server: {0}", e.Message));
	}

	return verNew;
}
```

**Comment by neoh4x0r** [comment]
@DuskDweller 
@squid-box 

Here is some test code that tests the following urls (both using mono and wine) without logging in.

The **ExtendedWebClient** in **NexusClient/Updating/ProgrammeUpdater.cs**
should probably be updated with the code included in the zip (the **ServiceClient** class).

This code should work as-is without needing the authentication token(s) where some function (like toggle endorsements) requires you to be logged in.

For any function that requires that the user be logged in will need to call the appropriate api methods from INexusModRepositoryApi.

**Latest version:**
https://legacy-api.nexusmods.com/NMM?GetLatestVersion

**Get Mod Info (SkyUI)**
https://legacy-api.nexusmods.com/skyrim/Mods/3863/?game_id=110

[nexus-api-test.zip](https://github.com/Nexus-Mods/Nexus-Mod-Manager/files/2065491/nexus-api-test.zip)


The output of the test:

```
ServiceClient::CreateRequest::['https://legacy-api.nexusmods.com/NMM?GetLatestVersion']
ServiceClient::GetResponse:: [ContentType:'text/html'] [StatusCode: 'OK']
LATEST_VERSION_URL (TEST Result): '0.63.14|https://legacy-api.nexusmods.com/Releases/4.5/Nexus Mod Manager-0.63.14.exe'
ServiceClient::CreateRequest::[original uri: 'https://legacy-api.nexusmods.com/skyrim/Mods/3863/?game_id=110']
ServiceClient::CreateRequest::[new uri: 'https://legacy-api.nexusmods.com/skyrim/Mods/3863/']
ServiceClient::CreateRequest::[post_data='game_id=110']
ServiceClient::GetResponse:: [ContentType:'application/json; charset=utf-8'] [StatusCode: 'OK']
SKYRIM_MOD_TEST_URL (TEST Result): '{"id":3863,"game_id":110,"name":"SkyUI","category_id":42,"summary":"Elegant, PC-friendly interface mod with many advanced features.","description":"[center][img]http:\/\/static-6.nexusmods.com\/15\/mods\/110\/images\/3575-0-1431941008.png[\/img][\/center]\r\n[center]SkyUI is an elegant, PC-friendly UI mod for Skyrim with many advanced features. All improvements seamlessly integrate with the style of the original interface. Version 5 includes better crafting, enchanting, alchemy and smithing menus and several other enhancements.[\/center]\r\n[center][b]Core team[\/b]: snakster, Mardoxx\r\n[b]Artwork[\/b]: psychosteve[\/center]\r\n\r\n[center][img]http:\/\/static-5.nexusmods.com\/15\/mods\/110\/images\/3575-0-1431913254.png[\/img][\/center]\r\n[list]\r\n[*] Convenient inventory management with a feature-rich panel that supports full-text search, sorting and extended data columns.\r\n[\/list]\r\n[list]\r\n[*] [b]Version 5 extends this to all menus, including crafting, enchanting, alchemy and smithing.[\/b]\r\n[\/list]\r\n[list]\r\n[*] A powerful, categorized favorites menu that still keeps true to the original idea of a fast, non-intrusive way to access your essential items. Eight customizable hotkey groups allow to explore playstyles that were way too tedious before.\r\n[\/list]\r\n[list]\r\n[*] The Mod Configuration Menu (MCM) to support user-friendly in-game customization. It&#39;s not just used by SkyUI, but by hundreds of other mods as well.\r\n[\/list]\r\n[list]\r\n[*] Full-text search and highlighting of discovered locations on the map.\r\n[\/list]\r\n[list]\r\n[*] Active effects are displayed directly in the HUD, instead of having to constantly open the magic menu. Highly customizable.\r\n[\/list]\r\n[list]\r\n[*] Gamepad support, and intuitive and efficient keyboard navigation.\r\n[\/list]\r\n[list]\r\n[*] Multi-language support for English, French, German, Italian, Spanish, Russian, Polish, Czech and Japanese.\r\n[\/list]\r\n[list]\r\n[*] [url=http:\/\/steamcommunity.com\/sharedfiles\/filedetails\/?id=426512192]For a complete list, see the [i]Features Overview[\/i][\/url].\r\n[\/list]\r\n[list]\r\n[*] [url=https:\/\/www.youtube.com\/watch?v=9qzBT0ywHgk]For a visual tour, check out our latest trailer[\/url].\r\n[\/list]\r\n\r\n\r\n[center][img]http:\/\/static-2.nexusmods.com\/15\/mods\/110\/images\/3575-0-1431941212.png[\/img][\/center]\r\n[b][size=4][color=#7cbeff]STEP 1:[\/color] Install the Skyrim Script Extender (SKSE)[\/size][\/b]\r\n[list][\/list]\r\n[list]\r\n[*] [url=http:\/\/store.steampowered.com\/app\/365720]Install SKSE from Steam[\/url], or download the installer from [url=http:\/\/skse.silverlock.org\/]skse.silverlock.org[\/url] and run it.\r\n[*] Required SKSE version is 1.7.3 or newer.\r\n[\/list]\r\n\r\n[b][size=4][color=#7cbeff]STEP 2:[\/color] Install SkyUI[\/size][\/b]\r\n[list][\/list]\r\n[list]\r\n[*] [b][size=3]Option A - Nexus Mod Manager (NMM)[\/size][\/b]\r\n[list]\r\n[*] Click the [color=#7cbeff]Download (NMM)[\/color] button on top of this page.\r\n[*] [color=#7cbeff]Activate SkyUI[\/color] in NMM.\r\n[\/list][\/list]\r\n[list]\r\n[*] [b][size=3]Option B - Manual Installation[\/size][\/b]\r\n[list]\r\n[*] Locate the [color=#7cbeff]Data\/[\/color] folder in your Skyrim [color=#7cbeff]installation directory[\/color]. Typically it&#39;s found at [color=#7cbeff]Program Files\/Steam\/steamapps\/common\/skyrim\/Data\/[\/color].\r\n[*] Extract the contents of the downloaded archive file to your [color=#7cbeff]Data\/[\/color] folder.\r\n[*] In the Skyrim Launcher, select [color=#7cbeff]Data Files[\/color] and enable [color=#7cbeff]SkyUI.esp[\/color].\r\n[\/list][\/list]\r\n\r\n\r\n[center][img]http:\/\/static-1.nexusmods.com\/15\/mods\/110\/images\/3575-0-1431912981.png[\/img][\/center]\r\n[list]\r\n[*] [url=http:\/\/steamcommunity.com\/sharedfiles\/filedetails\/?id=429510726]Questions about anything? Check out the [i]FAQ[\/i][\/url].\r\n[\/list]\r\n[list]\r\n[*] [url=http:\/\/steamcommunity.com\/sharedfiles\/filedetails\/?id=429076438]Did you run into a problem? Have a look at the [i]Troubleshooting Guide[\/i][\/url].\r\n[\/list]\r\n[list]\r\n[*] [url=https:\/\/github.com\/schlangster\/skyui\/wiki]Are you a mod author interested in using MCM? The [i]Documentation[\/i] explains how it works[\/url].\r\n[\/list]\r\n\r\n[center][img]http:\/\/static-6.nexusmods.com\/15\/mods\/110\/images\/3575-0-1431941429.png[\/img][\/center]\r\n[list]\r\n[*] [i]T3T[\/i], for creating the initial icon set that was used in earlier versions.\r\n[*] [i]ianpatt[\/i], for his continued support of SKSE.\r\n[*] [i]Kratos[\/i], for helping to get the project off ground when we started.\r\n[*] [i]Verteiron[\/i], for prototyping the Papyrus script for the favorites menu.\r\n[*] [i]behippo[\/i] and [i]PurpleLunchBox[\/i], for their SKSE contributions.\r\n[*] [i]Gibbed[\/i], for prototyping the inventory data extension and container categorization.\r\n[*] [i]GreatClone[\/i], for his set of category icons that was used in earlier versions.\r\n[*] [i]MisterPete[\/i], for coming up with a scheme for improved column sorting.\r\n[*] [i]Gopher[\/i], for helping with the NMM installer and promoting SkyUI on his YouTube channel.\r\n[*] [i]Indie[\/i], for creating our first trailer.\r\n[*] [i]Lojack[\/i], for improving integration with Wyre Bash.\r\n[*] [i]Ismelda[\/i], for providing config values for very high resolutions used with multi-monitor setups.\r\n[*] [i]Wakeupbrandon[\/i], for inspiring the development of this mod with his mockups.\r\n[*] [i]Pelinor[\/i], for originally developing MCM for Fallout: New Vegas and allowing us to use his logo.\r\n[*] [i]cdcooley[\/i], for going out of his way to answer user questions on the Nexus sites.\r\n[*] Thanks to all testers and users who provided feedback. You helped us a great deal with improving the quality of this mod. Also thanks to everyone who helped with the translations.\r\n[\/list]","version":"5.1","author":"SkyUI Team","lastupdate":"\/Date(1439906093000)\/","adult":0,"date":"\/Date(1324081488000)\/","OLD_mid":28794,"OLD_mname":"schlangster","OLD_downloads":18487948,"OLD_endorsements":637049,"OLD_size":1315,"OLD_image":"3863-0-1436879491.png","OLD_u_downloads":4758668,"OLD_views":38433234,"OLD_readme":"","allow_view":1,"allow_rating":1,"allow_tagging":1,"allow_images":0,"allow_topics":1,"perm_useinstructions":0,"perm_others":2,"perm_upload":0,"perm_convert":0,"perm_modify":2,"perm_use":2,"perm_notes":"","perm_credits":"Besides the SkyUI team itself, there are other people as well who helped significantly to make this mod a reality.\r\nIn the following they are listed by name, including a list of their contributions.\r\n\r\nKratos:\r\n\tWas a core member of the SkyUI team until version 2.1 and as such contributed significantly to the project in various areas.\r\n\r\nianpatt:\r\n\tAdded lots of new functions to the Skyrim Script Extender, that greatly helped us during development and enabled new features that\r\n\twould otherwise be impossible.\r\n\r\nbehippo:\r\n\tHelped improving\/advancing the interface extensions plugin by decoding the game classes and giving us access to them through SKSE.\r\n\r\nGibbed:\r\n\tCreated the 'gibbed interface extensions' SKSE plugin, which makes more game data available for display in the inventory.\r\n\tAlso allowed us to bundle his container categorization plugin. As of version 2.1, both these plugins have been included in SKSE\r\n\titself.\r\n\r\nIndie:\r\n\tCreated our trailer and helps with QA and user support.\r\n\r\nGreatClone:\r\n\tCreated an amazing set of alternative category icons.\r\n\r\nGopher:\r\n\tDid most of the work on the NMM installer, created an installation tutorial video and promoted SkyUI on his YouTube channel.\r\n\r\nLojack:\r\n\tCreated a BCF (including an installation wizard) for SkyUI to improve the installation experience for Wyre Bash users.\r\n\tAlso added an auto-conversion feature to Wyre Bash itself so this BCF is automatically applied.\r\n\r\nIsmelda:\r\n\tProvided configs for very high resolutions used with multi-monitor setups.\r\n\r\nWakeupbrandon:\r\n\tHis mock-up inspired the overall layout of the new inventory.\r\n\r\nMisterPete:\r\n\tImplemented new column types for improved sorting.\r\n\r\nPsychosteve:\r\n\tCreated our new primary icon theme for version 3.0.\r\n\tHe also created the Active Effect icons that have been added in version 3.2.\r\n\r\nPelinor:\r\n\tThe creator of MCM for Fallout: New Vegas. Allowed us to use his original logo.\r\n\r\n\r\nSkyUI is utilizing TweenLite, a high-performance tweening library (http:\/\/www.greensock.com\/tweenlite\/).\r\n\tThanks to Jack Doyle and his team for creating it and allowing us to use it under their \u201cNo Charge\u201d license.\r\n\r\nThanks to all the testers, who helped a great deal with improving the overall quality of this mod:\r\n\tToJKa, HellKnight, xporc, MadCat221, Ismelda, Gribbles, freesta, Cartrodus, TheCastle,\r\n\tNewRaven, T-qb, V4N0, Noritep, claudekennilol, dj2005, llfbandit, webrunner, 352, Erelde, tejon, Sagittarius22\r\n\t(in random order)\r\n\r\nThanks to all translators who helped localizing SkyUI to different languages, among them being:\r\n\tSagittarius22, LLFBandit, xporc (French)\r\n\txlwarrior, aloot (Spanish)\r\n\tStarfis (Czech)\r\n\tRhaegal, aloot (Italian)\r\n\tpatryk110 (Polish)\r\n\tvova2112 (Russian)\r\n\r\nLast but not least, thanks to the whole SKSE team, because without their Script Extender creating this mod wouldn't have been possible.","OLD_perm_use":2,"thread_id":499516,"lastcomment":"\/Date(1527979417000)\/","moderate":0,"pm_notify":18069,"donate":1,"lock_comments":0,"ip":"","published":1,"has_hot_image":1,"has_back_image":0,"commenting":2,"featured_count":0,"hidden_reason":"","wizard_steps":"a:8:{i:0;s:7:\"publish\";i:1;s:7:\"details\";i:2;s:5:\"media\";i:3;s:4:\"docs\";i:4;s:11:\"permissions\";i:5;s:12:\"requirements\";i:6;s:5:\"files\";i:7;s:8:\"articles\";}","suggested_category":0,"two_weeks_ratings":2625,"one_week_ratings":1374,"type":1,"language":5,"translation_of":0,"public_graphs":0,"allow_videos":2,"allow_bug_reports":0,"donate_popup":1,"perm_notformoney":1,"donate_endorsement":1,"abandoned_status":0,"NEW_downloads":6901236,"NEW_u_downloads":4758668,"console_intentions":3,"bethnet_xbox_id":0,"bethnet_ps_id":0,"donation_description":"","last_activity":"\/Date(1440427123000)\/","perm_earn_dp":0,"mod_page_uri":"https:\/\/www.nexusmods.com\/skyrim\/mods\/3863"}'
```


PS: I can visit both of the urls in my broswer and I get the data back if....
I used a browser-extension to set my user agent to: **Nexus Client v0.65.5**

**Comment by neoh4x0r** [comment]
@DuskDweller 
@squid-box 

The PR I just opened should fix this issue.
EDIT: I just pushed some more changes that implements github releases.
(at least grabbing the latest version and the download url for the installer)

However, it does seems like pulling github releases would be better.

**So instead of pulling from here:**
https://legacy-api.nexusmods.com/NMM?GetLatestVersion

**The latest release is listed as the first element in the json from the release api:**
https://api.github.com/repos/Nexus-Mods/Nexus-Mod-Manager/releases

**Example:**
https://api.github.com/repos/Nexus-Mods/Nexus-Mod-Manager/releases

Important keywords:
tag_name: 0.65.5
name: "Release 0.65.5"
"browser_download_url": "https://github.com/Nexus-Mods/Nexus-Mod-Manager/releases/download/0.65.5/Nexus.Mod.Manager-0.65.5.exe"
etc
```
[
  {
    "url": "https://api.github.com/repos/Nexus-Mods/Nexus-Mod-Manager/releases/11159055",
    "assets_url": "https://api.github.com/repos/Nexus-Mods/Nexus-Mod-Manager/releases/11159055/assets",
    "upload_url": "https://uploads.github.com/repos/Nexus-Mods/Nexus-Mod-Manager/releases/11159055/assets{?name,label}",
    "html_url": "https://github.com/Nexus-Mods/Nexus-Mod-Manager/releases/tag/0.65.5",
    "id": 11159055,
    "node_id": "MDc6UmVsZWFzZTExMTU5MDU1",
    "tag_name": "0.65.5",
    "target_commitish": "735fb4f5f07ccd1cf4a34ca0ac9615927b87183f",
    "name": "Release 0.65.5",
    "draft": false,
    "author": {
      "login": "DuskDweller",
      "id": 7575288,
      "node_id": "MDQ6VXNlcjc1NzUyODg=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/7575288?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/DuskDweller",
      "html_url": "https://github.com/DuskDweller",
      "followers_url": "https://api.github.com/users/DuskDweller/followers",
      "following_url": "https://api.github.com/users/DuskDweller/following{/other_user}",
      "gists_url": "https://api.github.com/users/DuskDweller/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/DuskDweller/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/DuskDweller/subscriptions",
      "organizations_url": "https://api.github.com/users/DuskDweller/orgs",
      "repos_url": "https://api.github.com/users/DuskDweller/repos",
      "events_url": "https://api.github.com/users/DuskDweller/events{/privacy}",
      "received_events_url": "https://api.github.com/users/DuskDweller/received_events",
      "type": "User",
      "site_admin": false
    },
    "prerelease": false,
    "created_at": "2018-05-24T14:38:54Z",
    "published_at": "2018-05-24T14:42:35Z",
    "assets": [
      {
        "url": "https://api.github.com/repos/Nexus-Mods/Nexus-Mod-Manager/releases/assets/7290216",
        "id": 7290216,
        "node_id": "MDEyOlJlbGVhc2VBc3NldDcyOTAyMTY=",
        "name": "Nexus.Mod.Manager-0.65.5.exe",
        "label": null,
        "uploader": {
          "login": "DuskDweller",
          "id": 7575288,
          "node_id": "MDQ6VXNlcjc1NzUyODg=",
          "avatar_url": "https://avatars0.githubusercontent.com/u/7575288?v=4",
          "gravatar_id": "",
          "url": "https://api.github.com/users/DuskDweller",
          "html_url": "https://github.com/DuskDweller",
          "followers_url": "https://api.github.com/users/DuskDweller/followers",
          "following_url": "https://api.github.com/users/DuskDweller/following{/other_user}",
          "gists_url": "https://api.github.com/users/DuskDweller/gists{/gist_id}",
          "starred_url": "https://api.github.com/users/DuskDweller/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/DuskDweller/subscriptions",
          "organizations_url": "https://api.github.com/users/DuskDweller/orgs",
          "repos_url": "https://api.github.com/users/DuskDweller/repos",
          "events_url": "https://api.github.com/users/DuskDweller/events{/privacy}",
          "received_events_url": "https://api.github.com/users/DuskDweller/received_events",
          "type": "User",
          "site_admin": false
        },
        "content_type": "application/x-msdownload",
        "state": "uploaded",
        "size": 6792909,
        "download_count": 4836,
        "created_at": "2018-05-24T14:41:46Z",
        "updated_at": "2018-05-24T14:41:55Z",
        "browser_download_url": "https://github.com/Nexus-Mods/Nexus-Mod-Manager/releases/download/0.65.5/Nexus.Mod.Manager-0.65.5.exe"
      }
    ],
    "tarball_url": "https://api.github.com/repos/Nexus-Mods/Nexus-Mod-Manager/tarball/0.65.5",
    "zipball_url": "https://api.github.com/repos/Nexus-Mods/Nexus-Mod-Manager/zipball/0.65.5",
    "body": "# Release Notes\r\n\r\nPlease be aware that these latest releases are not fully code signed as our certificate has elapsed - so you will need to accept any warnings that Windows throws at you.\r\n\r\n### 0.65.5\r\n- **New features**\r\n  - Added launch parameters to LOOT for supported games. (thanks to user squid-box on Github)\r\n- **Bugfixes**\r\n  - Fixed issue with illegal characters in file paths. (thanks to user kkriisgaard on Github)\r\n  - Updated 7z dlls. (thanks to user squid-box on Github)\r\n  - Fixed issue with unsupported file systems during game detection. (thanks to user squid-box on Github)\r\n  - Additional tracing for plugins.txt related issues. (thanks to user squid-box on Github)\r\n  - Additional tracing for Fallout 4 version retrieval issues. (thanks to user squid-box on Github)\r\n  - Added prompt when the program's config file is corrupt. (thanks to user squid-box on Github)\r\n\r\n\r\n"
  },
```

**Comment by dyslexicanaboko** [comment]
@squid-box was this ever resolved by any chance? It seems directly related to what I was working on most recently, issue 461.

**Comment by squid-box** [comment]
I believe this was circumvented by grabbing update info from Github instead, since the Nexus server won't go above 0.65.2 anyway.

**Comment by neoh4x0r** [comment]
@squid-box 
Yes, updating to pull from Github does resolve this single issue.

However, it is a far more reaching issue that just this.

Currently the legacy-api server will reject any requests (sends 403 forbidden) that do not have a user-agent of **Nexus Client v0.65.5**

I'm not sure what the channel factory sends as the user-agent when using NMM repository API, but it would need to be verified that it sends the proper one.

**Comment by dyslexicanaboko** [comment]
@neoh4x0r that is really good to know thank you. That is probably why everyone is having problems logging in. I still can't login and as far as I can tell it has nothing to do with the client side code. This gives me something to go on.

**Comment by neoh4x0r** [comment]
@dyslexicanaboko 

Before the code was updated to move to github release....

I had pushed this commit to set the user-agent in ExtendedWebClient
The same things probably needs to be done in web.config (but I am not sure what the syntax is for that).

https://github.com/Nexus-Mods/Nexus-Mod-Manager/pull/365/commits/5007d9dfbc808df97bba21a730a25e1ef96aa1c8

You can also refer to this comment I posted: nexus-api-test.zip
The second test attempts to grab mod info for SkyUI (id:3863)
without the user-agent being set the test failed.

https://github.com/Nexus-Mods/Nexus-Mod-Manager/issues/346#issuecomment-394124089

**Comment by squid-box** [comment]
@dyslexicanaboko: Just to disrupt the discussion, I've had no issues logging in to any version so far - so it doesn't seem to be a universal problem.

**Comment by dyslexicanaboko** [comment]
@squid-box my friend tells me the same thing, so definitely something weird on my end. I have no idea why I cannot logon. Maybe my password is too complex, that has bit me in the past. It's my email address and something like this: "eBDR._*8^E3.s2p"

Everytime I would try I would get an error like @neoh4x0r was talking about. I think it was a 403 but I am not sure. I will debug again and see what the deal is, just difficult when you are having a one sided conversation with a server you don't control.

**Comment by squid-box** [comment]
The most exotic characters in my password are ~ and `.
I don't use my email to log in though, I use my username (squidbox). No idea if that makes a difference, but I can't log in using my email.

**Comment by neoh4x0r** [comment]
@squid-box 
@dyslexicanaboko 

I had special characters in my password originally, but moved to a very long mix of letters and numbers.

I was still getting 403 errors when logging in (even after eliminating special characters)......so while the server doesn't like special characters being html encoded (it does not look like the password [unless it was wrong] is to blame for the 403 errors).

Although the user-agent does seem to be set:
https://github.com/Nexus-Mods/Nexus-Mod-Manager/blob/master/NexusClient/ModRepositories/Nexus/NexusModRepository.cs#L180

**Comment by dyslexicanaboko** [comment]
~@neoh4x0r @squid-box this is still broken for me - just a confirmation. I am going to be debugging, I think I have a better idea now of what might be going on.~

Ok I took a deeper look, used postman and I think I understand the confusion now. The username cannot be an email address even though the Nexus Mods website accepts it. For the purposes of NMM it has to be your username.

I reset my password and using postman I was able to login. I then was able to logon using NMM but specifically using a username not an email address.
